Firestore/core/test/unit/model/value_util_test.cc

```diff
@@ -152,57 +130,95 @@ class ValueUtilTest : public ::testing::Test {
   remote::Serializer serializer{DbId()};
 };
 
+TEST(FieldValueTest, ValueHelpers) {
+  // Validates that the Value helpers in testutil produce the right types
+  google_firestore_v1_Value bool_value = Value(true);
+  ASSERT_EQ(GetTypeOrder(bool_value), TypeOrder::kBoolean);
+  EXPECT_EQ(bool_value.boolean_value, true);
+
+  google_firestore_v1_Value int_value = Value(5);
+  ASSERT_EQ(GetTypeOrder(int_value), TypeOrder::kNumber);
+  EXPECT_EQ(int_value.integer_value, 5);
+
+  google_firestore_v1_Value long_value = Value(LONG_MAX);
```

Comment:
Optional: you can use `std::numeric_limits<long>::max()` to avoid using a macro (here and below).